### PR TITLE
Enforce documentation with swiftlint `missingDocs` rule

### PR DESCRIPTION
**Pull request by OdNairy**
* Adding `switflint` rule to generate warnings (errors on CI) for missing documentation symbols

MAPSIOS-303

## Pull request checklist:
 - [ ] Write tests for all new functionality. If tests were not written, please explain why.
 - [ ] Add documentation comments for any added or updated public APIs.
 - [ ] Add any new public, top-level symbols to the Jazzy config's `custom_categories` (scripts/doc-generation/.jazzy.yaml)
 - [ ] Describe the changes in this PR, especially public API changes.
 - [ ] Add a changelog entry to to bottom of the relevant section (typically the `## main` heading near the top).
 - [ ] Update the guides (internal access only), README.md, and DEVELOPING.md if their contents are impacted by these changes.
 - [ ] Review and agree to the Contributor License Agreement ([CLA](https://github.com/mapbox/mapbox-maps-ios/blob/main/CONTRIBUTING.md#contributor-license-agreement)).

MAPSIOS-303

### CODE REVIEW

**Review by evil159 — APPROVED** [review]

File: Sources/MapboxMaps/Annotations/AnnotationOrchestrator.swift [hunk 1]
```diff
@@ -1,6 +1,7 @@
 import UIKit
 @_implementationOnly import MapboxCommon_Private
 
+/// An top-level interface for annotations.
```
Comment (pjleonard37) [inline Sources/MapboxMaps/Annotations/AnnotationOrchestrator.swift:4]:
```suggestion
/// A top-level interface for annotations.
```

File: Sources/MapboxMaps/Foundation/Extensions/Core/RenderedQueryOptions.swift [hunk 2]
```diff
@@ -20,6 +20,7 @@ extension RenderedQueryOptions {
         self.init(__layerIds: layerIds, filter: filterJson)
     }
 
+    /// Filters the returned features with an expression.
```
Comment (pjleonard37) [inline Sources/MapboxMaps/Foundation/Extensions/Core/RenderedQueryOptions.swift:23]:
```suggestion
    /// Filters the returned features with an ``Expression``.
```

File: Sources/MapboxMaps/Foundation/Extensions/Turf/Feature.swift [hunk 3]
```diff
@@ -1,3 +1,4 @@
+/// An object representing a spatially bounded thing.
```
Comment (pjleonard37) [inline Sources/MapboxMaps/Foundation/Extensions/Turf/Feature.swift:1]:
```suggestion
/// An object representing a spatially bounded geometry.
```

File: Sources/MapboxMaps/Foundation/MapInitOptions.swift [hunk 4]
```diff
@@ -1,6 +1,10 @@
 import Foundation
 
+/// A protocol used to provide ``MapInitOptions`` when initializing a MapView with a Storyboard or
```
Comment (pjleonard37) [inline Sources/MapboxMaps/Foundation/MapInitOptions.swift:3]:
```suggestion
/// A protocol used to provide ``MapInitOptions`` when initializing a ``MapView`` with a Storyboard or
```

File: Sources/MapboxMaps/Foundation/MapInitOptions.swift [hunk 5]
```diff
@@ -1,6 +1,10 @@
...
+/// a nib.
 @objc public protocol MapInitOptionsProvider {
+    /// A method to be called when map view needs initialization options
```
Comment (pjleonard37) [inline Sources/MapboxMaps/Foundation/MapInitOptions.swift:6]:
```suggestion
    /// A method to be called when ``MapView`` needs initialization options
```

File: Sources/MapboxMaps/Foundation/MapInitOptions.swift [hunk 6]
```diff
@@ -1,6 +1,10 @@
...
+    /// - Returns: Initializations options for the map view.
```
Comment (pjleonard37) [inline Sources/MapboxMaps/Foundation/MapInitOptions.swift:7]:
```suggestion
    /// - Returns: Initializations options for the ``MapView``.
```

File: Sources/MapboxMaps/Foundation/Projection.swift [hunk 7]
```diff
@@ -1,7 +1,10 @@
 import CoreLocation
 
+/// A collection of [Spherical Mercator](http://docs.openlayers.org/library/spherical_mercator.html) projection methods.
```
Comment (pjleonard37) [inline Sources/MapboxMaps/Foundation/Projection.swift:3]: This link is dead; perhaps use this link instead: 
```suggestion
/// A collection of [Spherical Mercator](https://en.wikipedia.org/wiki/Web_Mercator_projection) projection methods.
```

File: Sources/MapboxMaps/Style/StyleEncodable.swift [hunk 8]
```diff
@@ -1,9 +1,16 @@
 import Foundation
+
+/// A protocol describing style-related entity that can be encoded.
```
Comment (pjleonard37) [inline Sources/MapboxMaps/Style/StyleEncodable.swift:3]:
```suggestion
/// A protocol describing a style-related entity that can be encoded.
```

File: Sources/MapboxMaps/Style/Types/Expression.swift [hunk 9]
```diff
@@ -1,7 +1,15 @@
 import Foundation
 
+/// An expression defines a formula for computing the value of any layout property, paint property,
```
Comment (pjleonard37) [inline Sources/MapboxMaps/Style/Types/Expression.swift:3]:
```suggestion
/// An ``Expression`` defines a formula for computing the value of any layout property, paint property,
```

**Review by pjleonard37 — APPROVED** [review]
Added a few minor changes and updated a link

**Pull request merged**